react-components/toolbar-filter/step1.md

```diff
@@ -1,4 +1,69 @@
-Toolbars can be included to allow a user to manage and manipulate a data set. Data can be presented in any valid presentation, a table, a list, or a data visualization (chart), for example. The PatternFly Toolbar component is a flexible layout system that accommodates a variety of configurations that can be adapted to your specific needs.
+Toolbars consist of content rows of components. Most toolbars will only be a single row, but each row needs to be housed in a `DataToolbarContent` component.
 
-In this tutorial, you will build a toolbar filter and attach it to a table. The image below is what we'll end up with when we are done.
-<img src="toolbar-filter/assets/final.png" alt="Completed toolbar filter with table" style="box-shadow: rgba(3, 3, 3, 0.2) 0px 1.25px 2.5px 0px;" />
+1) **Navigate to the `src` folder and open** `src/App.js`{{open}}
+
+2) **Set up the general structure of an empty toolbar and add the necessary imports**
+
+Copy the following code into the `App.js` file, replacing all of the content there:
+
+<pre class="file" data-filename="App.js" data-target="replace">
+import React from &quot;react&quot;;
+import &quot;@patternfly/react-core/dist/styles/base.css&quot;;
+import { Button } from &quot;@patternfly/react-core&quot;;
+import {
+  EditIcon,
+  CloneIcon,
+  SyncIcon,
+  FilterIcon
+} from &quot;@patternfly/react-icons&quot;;
+import {
+  DataToolbar,
+  DataToolbarContent,
+  DataToolbarItem,
+  DataToolbarGroup,
+  DataToolbarToggleGroup,
+  DataToolbarFilter
+} from &quot;@patternfly/react-core/dist/esm/experimental&quot;;
+import Input from &quot;./components/input&quot;;
+import CheckboxSelect from &quot;./components/checkboxSelect&quot;;
+import Kebab from &quot;./components/kebab&quot;;
+import DemoTable from &quot;./components/table&quot;;
+
+class ToolbarDemoApp extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    return (
+      &lt;React.Fragment&gt;
+        Toolbar Demo
+      &lt;/React.Fragment&gt;
+    );
+  }
+};
+
+export default ToolbarDemoApp;
+</pre>
+
+3) **Wrap the `Toolbar Demo` text inside `React.Fragment` in a `DataToolbar` component**
+
+The `DataToolbar` should include the property `id="tutorial-toolbar"`.
+
+4) **Wrap the `Toolbar Demo` text inside `DataToolbar` in a `DataToolbarContent` component**
```

Comment:
Now that the text has been wrapped in a DataToolbar component, I think you can change this step to be:

wrap the DataToolbar component added in step 3 in a DataToolbarContent component.

Reply:
hm... except it goes the other way around. We need them to wrap the TEXT that's already in the DataToolbar component in a DataToolbarContent component. DataToolbar is the outer most component inside the React.Fragment.

And this is hard to explain in steps, but this process is how the developer will build their toolbar. They build the DataToolbar, and then place DataTooblarContents as rows within it. 

What makes it confusing is the addition of the text 'Toolbar Demo'. But without that text, there would be no feedback to the user that anything is working. The preview screen will be empty. That's why I start with the text in the code. 

I could reorder the steps so that the text is wrapped in the DataToolbarContent first, and then the DataToolbarContent is wrapped in the DataToolbar, but then they will see an error in the preview between steps and this will not be natural development flow.

I think the best solution is to remove the 'Toolbar Demo' text so they are just putting components within components without having to wrap things in this sort of confusing manner. The only downside to this, there will be no preview for this step.